agent/proxycfg/state_test.go

```diff
@@ -305,6 +304,203 @@ func TestState_WatchesAndUpdates(t *testing.T) {
 		stages   []verificationStage
 	}
 
+	newConnectProxyCase := func(meshGatewayProxyConfigValue structs.MeshGatewayMode) testCase {
```

Comment:
I lifted the definition of the former "connect-proxy" case into a generator function so I could reuse the whole setup for different proxy-devel values of the mesh gateway mode without copypasta.